src/vs/editor/common/languages/defaultDocumentColorsComputer.ts

```diff
@@ -0,0 +1,150 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Color, HSLA } from 'vs/base/common/color';
+import { IPosition } from 'vs/editor/common/core/position';
+import { IRange } from 'vs/editor/common/core/range';
+import { IColor, IColorInformation } from 'vs/editor/common/languages';
+
+export interface IDocumentColorComputerTarget {
+	getValue(): string;
+	positionAt(offset: number): IPosition;
+	findMatches(regex: RegExp): RegExpMatchArray[];
+}
+
+function _parseCaptureGroups(captureGroups: IterableIterator<string>) {
+	const values = [];
+	for (const captureGroup of captureGroups) {
+		const parsedNumber = Number(captureGroup);
+		if (parsedNumber || parsedNumber === 0 && captureGroup.replace(/\s/g, '') !== '') {
+			values.push(parsedNumber);
+		}
+	}
+	return values;
+}
+
+function _toIColor(r: number, g: number, b: number, a: number): IColor {
+	return {
+		red: r / 255,
+		blue: b / 255,
+		green: g / 255,
+		alpha: a
+	};
+}
+
+function _findRange(model: IDocumentColorComputerTarget, match: RegExpMatchArray): IRange | undefined {
+	const index = match.index;
+	const length = match[0].length;
+	if (!index) {
+		return;
+	}
+	const startPosition = model.positionAt(index);
+	const endPosition = model.positionAt(index + length);
+	const range: IRange = {
+		startLineNumber: startPosition.lineNumber,
+		startColumn: startPosition.column,
+		endLineNumber: endPosition.lineNumber,
+		endColumn: endPosition.column
+	};
```

Comment:
model.positionAt is not cheap. We know that the match is on the same line, so this can be
```suggestion
	const range: IRange = {
		startLineNumber: startPosition.lineNumber,
		startColumn: startPosition.column,
		endLineNumber: startPosition.lineNumber,
		endColumn: startPosition.column + length
	};
```